util/data/msgencode.c

```diff
@@ -827,7 +830,13 @@ attach_edns_record_max_msg_sz(sldns_buffer* pkt, struct edns_data* edns,
 	rdatapos = sldns_buffer_position(pkt);
 	sldns_buffer_write_u16(pkt, 0); /* rdatalen */
 	/* write rdata */
-	for(opt=edns->opt_list; opt; opt=opt->next) {
+	for(opt=edns->opt_list_modules_out; opt; opt=opt->next) {
+		sldns_buffer_write_u16(pkt, opt->opt_code);
```

Comment:
Padding could be here too
```suggestion
		if (opt->opt_code == LDNS_EDNS_PADDING) {
			padding_option = opt;
			continue;
		}
		sldns_buffer_write_u16(pkt, opt->opt_code);
```